### PR TITLE
[UI] zone wizard: change edit traffic type form of VMware

### PR DESCRIPTION
**Pull request by utchoang**
### Description

Fixes https://github.com/apache/cloudstack/issues/5177

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### Screenshots (if appropriate):
image

### CODE REVIEW

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by utchoang** [comment]
@blueorangutan ui

**Review by shwstppr — APPROVED** [review]
code LGTM. Forms with different fields show correctly for VMware and other hypervisors.
May need testing with an appropriate env to verify edited traffic label.

**Comment by DK101010** [comment]
Unfortunately, I cannot create another zone for the test due to a lack of resources. Code LGTM, but I get this error with the switch types:
Warning: You cannot set a form field before rendering a field associated with the value. You can use getFieldDecorator(id, options) instead v-decorator="[id, options]" to register it before render.

**Comment by rhtyd** [comment]
Manual test necessary to merge this /cc @nvazquez @sureshanaparti

**Comment by nvazquez** [comment]
@davidjumani can you please test this PR?

File: ui/src/views/infra/zone/ZoneWizardPhysicalNetworkSetupStep.vue [hunk 1]
```diff
@@ -429,7 +470,15 @@ export default {
...
+          if (this.hypervisor === 'VMware') {
+            trafficInEdit.traffic.vSwitchName = values.vSwitchName
+            trafficInEdit.traffic.vlanId = values.vlanId
+            if (this.isAdvancedZone) {
+              trafficInEdit.traffic.vSwitchType = values.vSwitchType
+            }
+          } else {
+            trafficInEdit.traffic.label = values.trafficLabel
+          }
           this.trafficInEdit = null
```
Comment (shwstppr) [inline ui/src/views/infra/zone/ZoneWizardPhysicalNetworkSetupStep.vue:482]: @utchoang is this line of code correct? Are we emitting correct values here?

Comment (shwstppr) [reply ui/src/views/infra/zone/ZoneWizardPhysicalNetworkSetupStep.vue:482]: Ignore this ^^

**Review by shwstppr — APPROVED** [review]
LGTM.

Tested changes comparing API calls with old UI
New UI:
Screenshot from 2021-07-15 12-38-56

Old UI:
Screenshot from 2021-07-15 12-31-12

**Comment by tombudghu** [comment]
Hi all, we've come across an odd issue that might just have to do with this update. It is a VMware based setup. Configuring an advanced zone and defining multiple traffic types for some reason all of the vswitches are reset with values entered into the last configured. Imagine we configure vswitch0 for management traffic and then go on to configure a distributes switch (enabled using the appropriate parameter) for say guest traffic, it also changes the previously configured vswitch0. This behaviour we have not seen in 4.15. Any ideas, please?

**Pull request merged**